tests/test_remote_config.py

```diff
@@ -0,0 +1,747 @@
+        )
+        server_config = server_template.evaluate()
+        assert server_config.get_boolean('promo_enabled') == default_config.get('promo_enabled')
+        assert server_config.get_int('promo_discount') == default_config.get('promo_discount')
+
+    def test_evaluate_default_when_no_default_value(self):
+        app = firebase_admin.get_app()
+        default_config = {'default_value': 'local default'}
+        template_data = SERVER_REMOTE_CONFIG_RESPONSE
+        template_data['parameters'] = {
+            'default_value': {}
+        }
+        server_template = remote_config.init_server_template(
+            app=app,
+            default_config=default_config,
+            template_data=ServerTemplateData('etag', template_data)  # Use ServerTemplateData here
+        )
+        server_config = server_template.evaluate()
+        assert server_config.get_string('default_value') == default_config.get('default_value')
+
+    def test_evaluate_default_when_in_default(self):
+        app = firebase_admin.get_app()
+        template_data = SERVER_REMOTE_CONFIG_RESPONSE
+        template_data['parameters'] = {
+            'remote_default_value': {}
+        }
+        default_config = {
+            'inapp_default': '🐕'
+        }
+        server_template = remote_config.init_server_template(
+            app=app,
+            default_config=default_config,
+            template_data=ServerTemplateData('etag', template_data)  # Use ServerTemplateData here
+        )
+        server_config = server_template.evaluate()
+        assert server_config.get_string('inapp_default') == default_config.get('inapp_default')
+
+    def test_evaluate_default_when_defined(self):
+        app = firebase_admin.get_app()
+        template_data = SERVER_REMOTE_CONFIG_RESPONSE
+        template_data['parameters'] = {}
+        default_config = {
+            'dog_type': 'shiba'
+        }
+        server_template = remote_config.init_server_template(
+            app=app,
+            default_config=default_config,
+            template_data=ServerTemplateData('etag', template_data)  # Use ServerTemplateData here
+        )
+        server_config = server_template.evaluate()
+        assert server_config.get_value('dog_type').as_string() == 'shiba'
+        assert server_config.get_value('dog_type').get_source() == 'default'
+
+    def test_evaluate_return_numeric_value(self):
+        app = firebase_admin.get_app()
+        template_data = SERVER_REMOTE_CONFIG_RESPONSE
+        default_config = {
+            'dog_age': 12
+        }
+        server_template = remote_config.init_server_template(
+            app=app,
+            default_config=default_config,
+            template_data=ServerTemplateData('etag', template_data)  # Use ServerTemplateData here
+        )
+        server_config = server_template.evaluate()
+        assert server_config.get_int('dog_age') == 12
+
+    def test_evaluate_return__value(self):
```

Comment:
Uber nit: duplicate underscore in the test name (return__value)?

Reply:
changed return__value to return_boolean_value